refactor: Replaces all hardcoded strings with @string resource (#412)

coderaashir opened the pull request:
This  is done only for the Activity and Fragment files and all XML files.

therajanmaurya commented on mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/chargedialog/ChargeDialogFragment.java:
```diff
@@ -239,7 +239,7 @@ public void onNothingSelected(AdapterView<?> parent) {
     @Override
     public void showChargesCreatedSuccessfully(Charges changes) {
-        Toast.makeText(getActivity(), "Charge created successfully", Toast.LENGTH_LONG)
+        Toast.makeText(getActivity(), R.string.change_created_success, Toast.LENGTH_LONG)
```
change to ```charge_created_successfully```

therajanmaurya commented on mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/loanchargedialog/LoanChargeDialogFragment.java:
```diff
@@ -247,7 +247,7 @@ public void onNothingSelected(AdapterView<?> parent) {
     @Override
     public void showLoanChargesCreatedSuccessfully(Charges charges) {
-        Toast.makeText(getActivity(), "Charge created successfully", Toast
+        Toast.makeText(getActivity(), R.string.charge_created, Toast
```
Use above string resources "charge_created_successfully"

coderaashir replied:
I changed it to "charge_created", I hope thats fine

therajanmaurya replied:
Still, there are two string resources with same value 
1. ```<string name="charge_created_success">Charge created successfully</string>```
2. ```<string name="charge_created">Charge created successfully</string>```

Stay with 1st string and use this.

therajanmaurya commented on mifosng-android/src/main/java/com/mifos/mifosxdroid/online/clientcharge/ClientChargeFragment.java:
```diff
@@ -213,7 +213,7 @@ public void showChargesList(Page<Charges> chargesPage) {
             if (chargesPage.getPageItems().size() == 0 &&
                     (chargesPage.getTotalFilteredRecords() == chargesList.size()))
-                Toaster.show(rootView, "No more Charges Available");
+                Toaster.show(rootView, getString(R.string.no_more_changes_available));
```
String name should be ```no_more_charges_available``` not  ```no_more_changes_available```

coderaashir commented on mifosng-android/src/main/res/layout/row_collection_list_group_client.xml:
```diff
@@ -18,7 +18,7 @@
             android:layout_width="0dp"
             android:layout_height="wrap_content"
-            android:text="237"
+            android:text="@string/_237"
```
client_name has been defined as "Client Name", not "Raj Kumar Kaushikl"

therajanmaurya commented:
PR looks Good there is one things need to change.
1. change commit message to "refactor: your message". you can use git command 
```git commit --amend```

@ishan1604 : @coderaashir have committed the main master branch two commits in this PR.
What you think, he has to rebase his branch or this can be fine for now.

coderaashir commented:
@therajanmaurya commit message has been changed
what do you mean by 2 commits? I can only see one, i rebased and squashed all the additional commits some time back

therajanmaurya commented:
I mean commit message, not PR message, and check your changed files. you have added the two more commits of main master branch, In file DashboardActivity.java and CenterListFragment.java.

coderaashir commented:
@therajanmaurya I've changed both messages 
Should I remove that commit? I updated my branch after I had made the PR, i think

therajanmaurya commented:
Still commit message is ```Puts all hardcore style strings in resources folder... ```
If you will change the messge then it will look like this
```refactor: Puts all hardcore style strings in resources folder... ```

coderaashir commented:
Sorry, I was running the command from another branch @therajanmaurya

ishan1604 commented on mifosng-android/src/main/res/values/strings.xml:
```diff
@@ -137,6 +137,40 @@
+    <string name="survey_name">Survey Name</string>
+    <string name="start_tracking">Start tracking</string>
+    <string name="stop_tracking">Stop tracking</string>
+    <string name="client_id">237</string>
```
I don't quite like this string here. 

If this is just for debugging purposes.

You should use ```tools:text``` attribute and you can hardcode your string there

ishan1604 commented on mifosng-android/src/main/res/values/strings.xml:
```diff
@@ -137,6 +137,40 @@
+    <string name="start_tracking">Start tracking</string>
+    <string name="stop_tracking">Stop tracking</string>
+    <string name="client_id">237</string>
+    <string name="_12_02_2014">12-02-2014</string>
```
Same problem with this string

coderaashir commented:
@therajanmaurya regarding the conflics: should I resolve them myself or should I consult with you?

coderaashir commented:
Like in MifosBaseFragment, for some I'm getting a build error in line 73: 

SavingsAccountSummaryFragment savingsAccountSummaryFragment
                = SavingsAccountSummaryFragment.newInstance(419, depositType);

(you can find more info in gitter)
 
So what should I pass as the boolean in the arguments?

therajanmaurya commented:
@ishan1604 Please the final look and merge

luckyman20 commented:
@coderaashir Please squash all the commits and solve the conflicts

droidchef commented:
The PR cannot be merged due to conflicts. They must be resolved before we can review the PR again and eventually merge it.
